lib-multisrc/greenshit/src/eu/kanade/tachiyomi/multisrc/greenshit/GreenShit.kt

```diff
@@ -105,11 +107,18 @@ abstract class GreenShit(
         val url = "$apiUrl/obras/novos-capitulos".toHttpUrl().newBuilder()
             .addQueryParameter("pagina", page.toString())
             .addQueryParameter("limite", "24")
-            .addQueryParameter("gen_id", "4")
+            .addQueryParameterIf(targetAudience != TargetAudience.All, "gen_id", targetAudience.toString())
             .build()
         return GET(url, headers)
     }
 
+    private fun HttpUrl.Builder.addQueryParameterIf(condition: Boolean, name: String, value: String): HttpUrl.Builder {
+        if (condition) {
+            addQueryParameter(name, value)
+        }
+        return this
+    }
```

Comment:
Might be too terse, but consider:
```suggestion
    private fun HttpUrl.Builder.addQueryParameterIf(condition: Boolean, name: String, value: String): HttpUrl.Builder =
        apply { if (condition) addQueryParameter(name, value) }
```